app/src/main/java/ca/klapstein/baudit/data/CareProvider.java

```diff
@@ -8,6 +8,13 @@
 
     private PatientTreeSet assignedPatientTreeSet;
 
+    public CareProvider(Username username, Password password, ContactInfo contactInfo){
+        this.setUsername(username);
```

Comment:
It may be useful to have the `Account` constructor handle setting the `username`, `password`, and `contactInfo`.

Look into how you can call `super(<init args>)` in java to simplify constructors among subclasses.